resources/js/features/root/components/header/FacilityControl.tsx

```diff
@@ -49,45 +35,37 @@ export const FacilityControl: Component = () => {
   );
 };
 
+// TODO: it just works, may be wrong. Maybe there is a better way of handling 'global mutable state' :D
 const useFacilityControl = () => {
   const params = useParams();
   const queryClient = useQueryClient();
 
   const facilitiesQuery = createQuery(() => System.facilitiesQueryOptions());
   const statusQuery = createQuery(() => User.statusQueryOptions());
 
-  const facilities = createMemo(() =>
-    facilitiesQuery.data?.filter((facility) =>
-      statusQuery.data?.members.find(
-        (member) => member.facilityId === facility.id
-      )
-    )
+  const facilities = createMemo(
+    () =>
+      facilitiesQuery.data?.filter(
+        (facility) => statusQuery.data?.members.find((member) => member.facilityId === facility.id),
+      ),
   );
 
   createEffect(() => {
     if (params.facilityUrl) {
-      const facilities = queryClient.getQueryData<FacilityResource[]>(
-        System.facilitiesQueryOptions().queryKey
-      );
-      const facility = facilities?.find(
-        (facility) => facility.url === params.facilityUrl
-      );
+      const facilities = queryClient.getQueryData<FacilityResource[]>(System.facilitiesQueryOptions().queryKey);
+      const facility = facilities?.find((facility) => facility.url === params.facilityUrl);
 
-      console.log(facility);
       if (!facility) return;
 
-      const status = queryClient.getQueryData<User.GetStatusData>(
-        User.keys.status()
-      );
+      const status = queryClient.getQueryData<User.GetStatusData>(User.keys.status());
 
-      if (status?.members.find((member) => member.facilityId === facility.id))
-        setFacilityId(facility.id);
+      if (status?.members.find((member) => member.facilityId === facility.id)) setActiveFacilityId(facility.id);
     }
   });
 
   createEffect(() => {
-    if (facilities() && facilities()!.length === 1)
-      setFacilityId(facilities()!.at(0)?.id);
+    const facilitiesSub = facilities();
+    if (facilitiesSub?.length === 1) setActiveFacilityId(facilitiesSub.at(0)?.id);
```

Comment:
Jak dasz [0] zamiast .at(0) to chyba nie musisz dawać ? :)

Reply:
.at specjalnie zwraca Type || undefined, tablica może być równie dobrze pusta. Jak zrobisz array[0].field i array jest puste, to się wywali

Reply:
i tak, wiem że tutaj sprawdzam length i może tutaj ma to sens, ale to po prostu przyzwyczajenie

Reply:
No, wiem jak to działa. Po prostu dla mnie jak piszesz `?` to znaczy że zakładasz możliwość że to jest undefined, a tutaj tak nie jest a ktoś patrząc na kod może się zmylić na moment, dlatego lepiej tak nie pisać `?` tylko jak już to `!` a najlepiej `[0]`, ale powiedzmy że tak naprawdę dosyć wszystko jedno :)

Reply:
a nie, jednak ts jest sprytniejszy, array[0] też zwraca Type || undefined